src/nvim/eval.c

```diff
@@ -1275,116 +1275,80 @@ int get_spellword(list_T *const list, const char **ret_word)
 
 
 // Call some vim script function and return the result in "*rettv".
-// Uses argv[argc] for the function arguments.  Only Number and String
-// arguments are currently supported.
+// Uses argv[argc-1] for the function arguments. argv[argc]
+// should have type VAR_UNKNOWN.
 //
 // Return OK or FAIL.
 int call_vim_function(
     const char_u *func,
     int argc,
-    const char_u *const *const argv,
-    bool safe,                       // use the sandbox
-    int str_arg_only,               // all arguments are strings
+    typval_T *argv,
+    int safe,                       // use the sandbox
     typval_T *rettv
```

Comment:
in the vim patch, `safe` is the last arg. what's the reason for changing the order? 

Also `safe` could stay as `bool`.